### PR TITLE
Add bits not command

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Add ~`math bit-not`~ `bits not` command, so `nu` support more bit relative operations

# Tests

Make sure you've done the following:

- [X] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [X] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [X] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: crates/nu-command/src/math/bit_not.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        let head = call.head;
+        input.map(
+            move |value| operate(value, head),
+            engine_state.ctrlc.clone(),
+        )
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Apply the floor function to a list of numbers",
```
Comment (sholderbach) [inline crates/nu-command/src/math/bit_not.rs:41]: Mismatching description

File: crates/nu-command/src/math/bit_not.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                ],
+                span: Span::test_data(),
+            }),
+        }]
+    }
+}
+
+fn operate(value: Value, head: Span) -> Value {
+    match value {
+        Value::Int { val, span } => Value::Int { val: !val, span },
```
Comment (sholderbach) [inline crates/nu-command/src/math/bit_not.rs:57]: This assumes the underlying representation should always be `i64` with two's complement arithmetic for negative values.

**Review by sholderbach — COMMENTED** [review]
I am not as aware about usecases for signed integers but would extend the universal programming language aspect of nu

**Comment by hustcer** [comment]
`math bit-not` / `bits not` / `bit not` which one is better? Are there any bit operations can't be classified into math ? I think so.

**Comment by fdncred** [comment]
I really don't like having some operators do bit commands and some sub commands doing bit commands. It makes figuring out how to do them difficult from a discoverability point of view. I'd still prefer them all to be in one place. Like the bits command I documented or if we're going to put them in math, then that's ok too. I just want to make all of them easier to find.

**Comment by hustcer** [comment]
@fdncred Agreed, we can add other bit commands later

**Comment by WindSoilder** [comment]
> I really don't like having some operators do bit commands and some sub commands doing bit commands. It makes figuring out how to do them difficult from a discoverability point of view. I'd still prefer them all to be in one place. Like the bits command I documented or if we're going to put them in math, then that's ok too. I just want to make all of them easier to find.

Sorry, I'll move it to a new bits folder.  Currently I put it in math just because we only have one bit relative command, others are operators

**Comment by fdncred** [comment]
@WindSoilder I understand. I'm just saying I'd like to have all the bits things consolidated eventually. I'm not sure having it in a separate folder helps. I think we can have operators but I'm still a fan of having a bits command as well. It could be similar to the bytes commands you wrote. It's probably fine leaving it here for now but I want to get back to it. I like to keep things as organized and as discoverable as possible.

**Comment by hustcer** [comment]
> Sorry, I'll move it to a new bits folder. Currently I put it in math just because we only have one bit relative command, others are operators

Thanks, and I will add the other bit commands later, from the bit operators I have added before.

**Comment by fdncred** [comment]
@hustcer This is where I documented the `bits` command, although we can add or take away from this https://gist.github.com/fdncred/6f5fb149c014c6de039401a6a07d3046

File: crates/nu-command/src/math/bit_not.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/math/bit_not.rs]: Oh, good catch, fix the description

File: crates/nu-command/src/math/bit_not.rs (hunk 2 above)
Comment (WindSoilder) [reply crates/nu-command/src/math/bit_not.rs]: Thanks, that's something I never thought about.

To handle this, I've add `--signed`, so user can make their input integer as a signed integer
And `--number-size` flag, to make user tell `nu` treat a unsigned number as `1-bytes/2-bytes/4-bytes/8-bytes number`

**Comment by fdncred** [comment]
are we ready to land this or are we still tweaking it?

File: crates/nu-command/src/math/bit_not.rs (hunk 2 above)
Comment (sholderbach) [reply crates/nu-command/src/math/bit_not.rs]: Forgot my two's complement math. For signed integers the size shouldn't matter as a larger datatype would result in the same sign extension. Word size only matters for the unsigned case

Comment (WindSoilder) [reply crates/nu-command/src/math/bit_not.rs]: yeah, this is current implementation

Comment (sholderbach) [reply crates/nu-command/src/math/bit_not.rs]: Can we maybe replace `--number-size` with `--num-bytes` else it could also be bits. Apart from that it seems ready to land and test for me

Comment (WindSoilder) [reply crates/nu-command/src/math/bit_not.rs]: Ok, renamed it

**Comment by WindSoilder** [comment]
I think it's ready to land, but it's better to ask @sholderbach have a final check again

**Comment by sholderbach** [comment]
Happy with it
